pkg/networkservice/common/clientinfo/client.go

```diff
@@ -0,0 +1,46 @@
+package clientinfo
+
+import (
+	"context"
+	"os"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+var names = map[string]string{"NODE_NAME": "NodeNameKey", "POD_NAME": "PodNameKey", "CLUSTER_NAME": "ClusterNameKey"}
+
+type clientInfo struct{}
+
+// NewClient - creates a new networkservice.NetworkServiceClient chain element that adds pod, node and cluster names to request from corresponding environment variables
+func NewClient() networkservice.NetworkServiceClient {
+	return &clientInfo{}
+}
+
+func (a *clientInfo) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	conn := request.GetRequestConnection()
+	if conn.Labels == nil {
+		conn.Labels = make(map[string]string)
+	}
+	for envName, labelName := range names {
+		value, exists := os.LookupEnv(envName)
+		if exists {
+			oldValue, isPresent := conn.Labels[labelName]
+			if isPresent {
+				logrus.Warningf("The label %s was already assigned to %s. Overwriting.", labelName, oldValue)
```

Comment:
Do we really need to replace values here?

Reply:
Actually i don't know. Maybe @edwarnicke knows which option is more suitable?